bb_wrapper/wrapper/request.py

```diff
@@ -1,13 +1,38 @@
+from time import sleep
 from json.decoder import JSONDecodeError
 
 import requests
 
+from requests import ConnectionError
+from urllib3.exceptions import ProtocolError
+
 from ..utils import _get_logger
 
 
 logger = _get_logger("requests")
 
 
+def retry_request(counter=5):
+    retry_request.counter = counter
+
+    def wrapper(http_method, *args, **kwargs):
+        from functools import wraps
+
+        @wraps(http_method)
+        def inner(self, *args, **kwargs):
+            try:
+                return http_method(self, *args, **kwargs)
+            except (ConnectionError, ConnectionResetError, ProtocolError):
+                if retry_request.counter > 0:
+                    retry_request.counter -= 1
+                    return inner(self, *args, **kwargs)
+                raise
+
+        return inner
+
+    return wrapper
+
```

Comment:
Precisa mesmo da complexidade de usar o wraps do functools?

E passar explicitamente o self? Ele já seria o primeiro arg sem definir ele como um arg da função `inner` 


```suggestion
def retry_request(counter=5):
    retry_request.counter = counter

    def wrapper(func, *args, **kwargs):

        def inner(*args, **kwargs):
            try:
                return func(*args, **kwargs)
            except (ConnectionError, ConnectionResetError, ProtocolError):
                retry_request.counter -= 1
                if retry_request.counter > 0:
                    return inner(*args, **kwargs)
                raise

        return inner

    return wrapper

```

Reply:
Eu tinha colocado pra seguir o tutorial, mas vou testar sem.